
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Introduction to Flux Balance Analysis and Metabolic Engineering &#8212; ENGRI 1120</title>
    
  <!-- Loaded before other Sphinx assets -->
  <link href="../_static/styles/theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">
<link href="../_static/styles/pydata-sphinx-theme.css?digest=1999514e3f237ded88cf" rel="stylesheet">

    
  <link rel="stylesheet"
    href="../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" href="../_static/styles/sphinx-book-theme.css?digest=5115cc725059bd94278eecd172e13a965bf8f5a9" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../_static/proof.css" />
    <link rel="stylesheet" type="text/css" href="../_static/design-style.b7bb847fb20b106c3d81b95245e65545.min.css" />
    
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf">

    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/clipboard.min.js"></script>
    <script src="../_static/copybutton.js"></script>
    <script src="../_static/scripts/sphinx-book-theme.js?digest=9c920249402e914e316237a7dbc6769907cce411"></script>
    <script>let toggleHintShow = 'Click to show';</script>
    <script>let toggleHintHide = 'Click to hide';</script>
    <script>let toggleOpenOnPrint = 'true';</script>
    <script src="../_static/togglebutton.js"></script>
    <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../_static/design-tabs.js"></script>
    <script>const THEBE_JS_URL = "https://unpkg.com/thebe@0.8.2/lib/index.js"
const thebe_selector = ".thebe,.cell"
const thebe_selector_input = "pre"
const thebe_selector_output = ".output, .cell_output"
</script>
    <script async="async" src="../_static/sphinx-thebe.js"></script>
    <script>window.MathJax = {"options": {"processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="References" href="../References.html" />
    <link rel="prev" title="Introduction to Chemical Reaction Equilibrium" href="chemical-equilibrium.html" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="None">
    

    <!-- Google Analytics -->
    
  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="60">
<!-- Checkboxes to toggle the left sidebar -->
<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation" aria-label="Toggle navigation sidebar">
<label class="overlay overlay-navbar" for="__navigation">
    <div class="visually-hidden">Toggle navigation sidebar</div>
</label>
<!-- Checkboxes to toggle the in-page toc -->
<input type="checkbox" class="sidebar-toggle" name="__page-toc" id="__page-toc" aria-label="Toggle in-page Table of Contents">
<label class="overlay overlay-pagetoc" for="__page-toc">
    <div class="visually-hidden">Toggle in-page Table of Contents</div>
</label>
<!-- Headers at the top -->
<div class="announcement header-item noprint"></div>
<div class="header header-item noprint"></div>

    
    <div class="container-fluid" id="banner"></div>

    

    <div class="container-xl">
      <div class="row">
          
<!-- Sidebar -->
<div class="bd-sidebar noprint" id="site-navigation">
    <div class="bd-sidebar__content">
        <div class="bd-sidebar__top"><div class="navbar-brand-box">
    <a class="navbar-brand text-wrap" href="../index.html">
      
        <!-- `logo` is deprecated in Sphinx 4.0, so remove this when we stop supporting 3 -->
        
      
      
      <img src="../_static/cornell_seal_simple_black.svg" class="logo" alt="logo">
      
      
      <h1 class="site-logo" id="site-title">ENGRI 1120</h1>
      
    </a>
</div><form class="bd-search d-flex align-items-center" action="../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item active">
        
        <ul class="nav bd-sidenav bd-sidenav__home-link">
            <li class="toctree-l1">
                <a class="reference internal" href="../intro.html">
                    ENGRI 1120 Introduction to Chemical and Biomolecular Engineering
                </a>
            </li>
        </ul>
        <ul class="current nav bd-sidenav">
 <li class="toctree-l1">
  <a class="reference internal" href="../course.html">
   Course policies and procedures
  </a>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../julia-dir/julia-landing.html">
   Getting Started with Julia
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/>
  <label for="toctree-checkbox-1">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../julia-dir/julia-installation.html">
     Julia Installation
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../julia-dir/julia-basics.html">
     Introduction to Julia Basics
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../julia-dir/julia-data.html">
     Working with Data in Julia
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../chapter-1-dir/unit-1-introduction.html">
   Unit 1 Balances
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/>
  <label for="toctree-checkbox-2">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../chapter-1-dir/stuff-balances.html">
     Systems, Surroundings and Balance Equations
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../chapter-1-dir/material-balances.html">
     Material Balances
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../chapter-1-dir/energy-balances.html">
     Energy Balances
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../chapter-1-dir/money-balances.html">
     Financial Balances and Abstract Assets
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 has-children">
  <a class="reference internal" href="../chapter-2-dir/chapter-2-intro.html">
   Unit 2 Phases
  </a>
  <input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/>
  <label for="toctree-checkbox-3">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul>
   <li class="toctree-l2">
    <a class="reference internal" href="../chapter-2-dir/single-component-phase-eq.html">
     Understanding Phase Equilibrium
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../chapter-2-dir/vle.html">
     Vapor Liquid Equilibrium
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../chapter-2-dir/lle.html">
     Liquid Liquid Equilibrium
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1 current active has-children">
  <a class="reference internal" href="chapter-3-intro.html">
   Unit 3 Reactions
  </a>
  <input checked="" class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/>
  <label for="toctree-checkbox-4">
   <i class="fas fa-chevron-down">
   </i>
  </label>
  <ul class="current">
   <li class="toctree-l2">
    <a class="reference internal" href="chemical-kinetics.html">
     Introduction to Chemical Reaction Kinetics
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="chemical-equilibrium.html">
     Introduction to Chemical Reaction Equilibrium
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     Introduction to Flux Balance Analysis and Metabolic Engineering
    </a>
   </li>
  </ul>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="../References.html">
   References
  </a>
 </li>
</ul>

    </div>
</nav></div>
        <div class="bd-sidebar__bottom">
             <!-- To handle the deprecated key -->
            
            <div class="navbar_extra_footer">
            Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
            </div>
            
        </div>
    </div>
    <div id="rtd-footer-container"></div>
</div>


          


          
<!-- A tiny helper pixel to detect if we've scrolled -->
<div class="sbt-scroll-pixel-helper"></div>
<!-- Main content -->
<div class="col py-0 content-container">
    
    <div class="header-article row sticky-top noprint">
        



<div class="col py-1 d-flex header-article-main">
    <div class="header-article__left">
        
        <label for="__navigation"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="right"
title="Toggle navigation"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-bars"></i>
  </span>

</label>

        
    </div>
    <div class="header-article__right">
<button onclick="toggleFullScreen()"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="bottom"
title="Fullscreen mode"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>

<div class="menu-dropdown menu-dropdown-repository-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Source repositories">
      <i class="fab fa-github"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="https://github.com/varnerlab/ENGRI-1120-IntroToChemE-Book"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Source repository"
>
  

<span class="headerbtn__icon-container">
  <i class="fab fa-github"></i>
  </span>
<span class="headerbtn__text-container">repository</span>
</a>

      </li>
      
      <li>
        <a href="https://github.com/varnerlab/ENGRI-1120-IntroToChemE-Book/issues/new?title=Issue%20on%20page%20%2Fchapter-3-dir/flux-balance-analysis.html&body=Your%20issue%20content%20here."
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Open an issue"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-lightbulb"></i>
  </span>
<span class="headerbtn__text-container">open issue</span>
</a>

      </li>
      
    </ul>
  </div>
</div>

<div class="menu-dropdown menu-dropdown-download-buttons">
  <button class="headerbtn menu-dropdown__trigger"
      aria-label="Download this page">
      <i class="fas fa-download"></i>
  </button>
  <div class="menu-dropdown__content">
    <ul>
      <li>
        <a href="../_sources/chapter-3-dir/flux-balance-analysis.md"
   class="headerbtn"
   data-toggle="tooltip"
data-placement="left"
title="Download source file"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="headerbtn__text-container">.md</span>
</a>

      </li>
      
      <li>
        
<button onclick="printPdf(this)"
  class="headerbtn"
  data-toggle="tooltip"
data-placement="left"
title="Print to PDF"
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="headerbtn__text-container">.pdf</span>
</button>

      </li>
      
    </ul>
  </div>
</div>
<label for="__page-toc"
  class="headerbtn headerbtn-page-toc"
  
>
  

<span class="headerbtn__icon-container">
  <i class="fas fa-list"></i>
  </span>

</label>

    </div>
</div>

<!-- Table of contents -->
<div class="col-md-3 bd-toc show noprint">
    <div class="tocsection onthispage pt-5 pb-3">
        <i class="fas fa-list"></i> Contents
    </div>
    <nav id="bd-toc-nav" aria-label="Page">
        <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#introduction">
   Introduction
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#theory-of-flux-balance-analysis">
   Theory of Flux Balance Analysis
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#traditional-fba-problem-structure">
     Traditional FBA problem structure
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#canonical-linear-program">
     Canonical linear program
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#flux-balance-analysis-as-a-linear-program">
     Flux balance analysis as a linear program
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#alternative-approach">
     Alternative approach
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#mole-based-problem-formulation">
       Mole-based problem formulation
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#convex-pathway-analysis-advanced-topic">
       Convex pathway analysis (advanced topic)
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#summary">
   Summary
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
</ul>

    </nav>
</div>
    </div>
    <div class="article row">
        <div class="col pl-md-3 pl-lg-5 content-container">
            <!-- Table of contents that is only displayed when printing the page -->
            <div id="jb-print-docs-body" class="onlyprint">
                <h1>Introduction to Flux Balance Analysis and Metabolic Engineering</h1>
                <!-- Table of contents -->
                <div id="print-main-content">
                    <div id="jb-print-toc">
                        
                        <div>
                            <h2> Contents </h2>
                        </div>
                        <nav aria-label="Page">
                            <ul class="visible nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#introduction">
   Introduction
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#theory-of-flux-balance-analysis">
   Theory of Flux Balance Analysis
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#traditional-fba-problem-structure">
     Traditional FBA problem structure
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#canonical-linear-program">
     Canonical linear program
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#flux-balance-analysis-as-a-linear-program">
     Flux balance analysis as a linear program
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#alternative-approach">
     Alternative approach
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#mole-based-problem-formulation">
       Mole-based problem formulation
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#convex-pathway-analysis-advanced-topic">
       Convex pathway analysis (advanced topic)
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#summary">
   Summary
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#references">
   References
  </a>
 </li>
</ul>

                        </nav>
                    </div>
                </div>
            </div>
            <main id="main-content" role="main">
                
              <div>
                
  <section class="tex2jax_ignore mathjax_ignore" id="introduction-to-flux-balance-analysis-and-metabolic-engineering">
<h1>Introduction to Flux Balance Analysis and Metabolic Engineering<a class="headerlink" href="#introduction-to-flux-balance-analysis-and-metabolic-engineering" title="Permalink to this headline">#</a></h1>
<section id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">#</a></h2>
<p>Metabolic engineering is the practice of optimizing genetic and regulatory processes within cells to increase the cell’s production of a desired small molecule or macromolecule product, e.g., proteins of interest, and is a critical component of Biotechnology. There are two categories of biotechnology: industrial and medical. Metabolic engineering plays a vital role in both sectors. Industrial biotechnology is typically consumer-focused, e.g., components of consumer products such as detergents, food products, and small-molecule chemical feedstocks. On the other hand, medical biotechnology develops molecules for human (and animal) health applications, e.g., antibodies, therapeutic proteins, vaccines, etc.</p>
<p>The overall objective of a metabolic engineer is to maximize the <code class="docutils literal notranslate"><span class="pre">performance</span></code> of a metabolic network, e.g., develop a system that produces a desired product with the highest possible yield or at the maximum possible rate, etc. Toward this objective, metabolic engineers manipulate the biochemical networks that cells use to convert raw materials into molecules necessary for the cell’s survival. Metabolic engineering specifically seeks to:</p>
<ol class="simple">
<li><p>Mathematically model biochemical networks, calculate the yield (product divided by substrate) of valuable products, and identify parts of the network that constrain the production of these products of interest.</p></li>
<li><p>Use genetic engineering techniques to modify the biochemical network to relieve constraints limiting production. Metabolic engineers can then model the modified network to calculate the new product yield and identify new constraints (back to 1).</p></li>
</ol>
<p>Thus, a key component of metabolic engineering is the mathematical analysis of (bio)chemical reaction networks. Toward this challenge, we’ll introduce a widely used tool to model and analyze reaction networks: flux balance analysis. In particular, this lecture will:</p>
<ul class="simple">
<li><p>Introduce the <a class="reference internal" href="#content-references-theory-flux-balance-analysis"><span class="std std-ref">Theory of Flux Balance Analysis</span></a>, the <a class="reference internal" href="#content-references-flux-balance-analysis-traditional-structure"><span class="std std-ref">Traditional FBA problem structure</span></a> and an <a class="reference internal" href="#content-references-flux-balance-analysis-alternative-structure"><span class="std std-ref">Alternative approach</span></a> that uses mole balances and the open extent of reaction.</p></li>
</ul>
<hr class="docutils" />
</section>
<section id="theory-of-flux-balance-analysis">
<span id="content-references-theory-flux-balance-analysis"></span><h2>Theory of Flux Balance Analysis<a class="headerlink" href="#theory-of-flux-balance-analysis" title="Permalink to this headline">#</a></h2>
<p>Flux balance analysis (FBA) is a mathematical modeling and analysis approach that estimates the <em>intracellular</em> or <em>cell-free</em> reaction rates (which are called <em>fluxes</em>) throughout a <em>steady state</em> reaction network (units: <span class="math notranslate nohighlight">\(\star\)</span>mol/volume-time). However, while we are introducing flux balance analysis in the context of metabolic network analysis, FBA is a general tool that can be used to estimate <em>flows</em> through many different types of networks and graphs, e.g., social graphs, communication networks, or other types of problems that can be represented as a network of graphs.</p>
<p>Flux balance analysis is arguably the most widely used computational tool in metabolic engineering. However, there are alternatives to FBA, such as metabolic flux analysis (MFA), but these alternatives vary more in the solution approach than the structure of the estimation problem.</p>
<section id="traditional-fba-problem-structure">
<span id="content-references-flux-balance-analysis-traditional-structure"></span><h3>Traditional FBA problem structure<a class="headerlink" href="#traditional-fba-problem-structure" title="Permalink to this headline">#</a></h3>
<p>Suppose we have a system, which can be either open or closed, with a species set <span class="math notranslate nohighlight">\(\mathcal{M}\)</span> and a reaction set <span class="math notranslate nohighlight">\(\mathcal{R}\)</span>. Further, suppose that the system (or at least part of it) is at or near a steady state. Then, the objective of the flux balance analysis is to estimate the value of the reaction rates operating in the system using <a class="reference external" href="https://en.wikipedia.org/wiki/Linear_programming">linear programming</a>.</p>
<div class="proof remark admonition" id="remark-lp">
<p class="admonition-title"><span class="caption-number">Remark 8 </span> (What is linear programming?)</p>
<section class="remark-content" id="proof-content">
<p>Linear programming (LP) is a method to estimate the best outcome (such as maximum profit, lowest cost, or the best possible rate of production) using a mathematical model whose requirements are represented by linear relationships subject to linear constraints.</p>
<p><a class="reference external" href="https://en.wikipedia.org/wiki/Simplex_algorithm">The development of the simplex algorithm</a>, an efficient approach for solving linear programs, has led to LPs being used to solve problems in many engineering applications and other diverse industries such as banking, education, forestry, energy, and logistics.</p>
</section>
</div></section>
<section id="canonical-linear-program">
<h3>Canonical linear program<a class="headerlink" href="#canonical-linear-program" title="Permalink to this headline">#</a></h3>
<p>Linear programs maximize (or minimize) a <em>linear</em> objective function <span class="math notranslate nohighlight">\(\mathcal{O}\)</span> subject to <em>linear</em> constraints and bounds on the variables we are searching over.</p>
<p>Suppose we need to estimate some <span class="math notranslate nohighlight">\(M\)</span>-dimensional vector <span class="math notranslate nohighlight">\(\mathbf{x}\)</span>, e.g., the best possible allocation of some scarce resource such as cars to people in a ride-sharing application. Further, our problem can be represented as a system of linear equations. Then, we can estimate the <em>optimal</em> value for the variables <span class="math notranslate nohighlight">\(\mathbf{x}\)</span> by solving the linear program (canonical form):</p>
<div class="math notranslate nohighlight">
\[\begin{split}
\begin{eqnarray}
\text{maximize/minimize}~\mathcal{O} &amp;=&amp; \sum_{i=1}^{M} c_{i}x_{i}\\
\text{subject to}~\mathbf{Ax} &amp;\leq&amp;\mathbf{b}\\
\text{and}~x_{i}&amp;\geq&amp;{0}\qquad{i=1,2,\dots,M}
\end{eqnarray}
\end{split}\]</div>
<p>The components of <span class="math notranslate nohighlight">\(\mathbf{x}\)</span> are the variables to be determined, <span class="math notranslate nohighlight">\(c_{i}\)</span> are constant coefficients in the <em>linear</em> objective function <span class="math notranslate nohighlight">\(\mathcal{O}\)</span>,
<span class="math notranslate nohighlight">\(\mathbf{A}\)</span> is a <span class="math notranslate nohighlight">\(N\times{M}\)</span> matrix and <span class="math notranslate nohighlight">\(\mathbf{b}\)</span> is a <span class="math notranslate nohighlight">\(N\times{1}\)</span> (constant) vector.</p>
</section>
<section id="flux-balance-analysis-as-a-linear-program">
<h3>Flux balance analysis as a linear program<a class="headerlink" href="#flux-balance-analysis-as-a-linear-program" title="Permalink to this headline">#</a></h3>
<p>The problem we are trying to solve when performing flux balance analysis is to estimate the vector of <em>unknown</em> reaction rates in a metabolic network such that some objective is maximized, e.g., the maximum rate of production subject to material balance and reaction bounds constraints.</p>
<p>The flux balance analysis problem is typically encoded as a linear programming (LP) problem:</p>
<div class="proof definition admonition" id="defn-fba-problem">
<p class="admonition-title"><span class="caption-number">Definition 38 </span> (Flux balance analysis problem)</p>
<section class="definition-content" id="proof-content">
<p>Let a system consist of the species <span class="math notranslate nohighlight">\(\mathcal{M}\)</span> and reactions <span class="math notranslate nohighlight">\(\mathcal{R}\)</span>. Further, let <span class="math notranslate nohighlight">\(v_{i}\in\mathcal{R}\)</span> denote the value of reaction rate (flux) <span class="math notranslate nohighlight">\(i\)</span> (units: mol/volume-time). Finally, we are interested in the steady-state value for the unknown reaction rates.</p>
<p>Then, the flux balance analysis problem, whose solution provides estimates for the values of the unknown fluxes inside the steady-state system, is a linear programming problem that maximizes (minimizes) the objective function:</p>
<div class="math notranslate nohighlight">
\[\text{maximize/minimze}~\mathcal{O}=\sum_{i\in\mathcal{R}}c_{i}v_{i}\]</div>
<p>subject to the linear constraints:</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{eqnarray}
\mathbf{S}\mathbf{v} &amp;=&amp; \mathbf{0}\\
\mathcal{L}\leq&amp;\mathbf{v}&amp;\leq\mathcal{U}\\
\dots &amp;=&amp; \dots
\end{eqnarray}\end{split}\]</div>
<p>The matrix <span class="math notranslate nohighlight">\(\mathbf{S}\)</span> is the <span class="math notranslate nohighlight">\(\mathcal{M}\times\mathcal{R}\)</span> stoichiometric matrix, <span class="math notranslate nohighlight">\(c_{i}\)</span> denote the objective coefficients,
<span class="math notranslate nohighlight">\(\mathbf{v}\)</span> denotes the unknown flux vector (the unknown that we are trying to estimate), and
<span class="math notranslate nohighlight">\(\mathcal{L}\)</span> (<span class="math notranslate nohighlight">\(\mathcal{U}\)</span>) denote the permissible lower (upper) bounds on the <em>unknown</em> fluxes.</p>
</section>
</div><p>The first set of constraints in <a class="reference internal" href="#defn-fba-problem">Definition 38</a> are material constraints, while the second imparts thermodynamic and kinetic information into the calculation. Finally, there are potentially other types of constraints (both linear and nonlinear) not shown in <a class="reference internal" href="#defn-fba-problem">Definition 38</a> that may be necessary for specific applications encountered in practice.</p>
<p>Checkout out the following reference to understand better the different components of a flux balance analysis problem:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/labs/pmc/articles/PMC3108565/">Orth, J., Thiele, I. &amp; Palsson, B. What is flux balance analysis?. Nat Biotechnol 28, 245–248 (2010). https://doi.org/10.1038/nbt.1614</a></p></li>
</ul>
</section>
<section id="alternative-approach">
<span id="content-references-flux-balance-analysis-alternative-structure"></span><h3>Alternative approach<a class="headerlink" href="#alternative-approach" title="Permalink to this headline">#</a></h3>
<p>The standard flux balance analysis problem is written in <code class="docutils literal notranslate"><span class="pre">concentration</span></code> units, e.g., the metabolic reaction flux has units of <span class="math notranslate nohighlight">\(\star\)</span>mol/volume-time. However, there is nothing that says we have to do that. For example, it may be more convenient to work in mole units instead. In this case, we estimate the unknown open reaction extent instead of calculating the unknown reaction rate.</p>
<section id="mole-based-problem-formulation">
<h4>Mole-based problem formulation<a class="headerlink" href="#mole-based-problem-formulation" title="Permalink to this headline">#</a></h4>
<p>We know from our <a class="reference internal" href="../chapter-1-dir/material-balances.html"><span class="doc std std-doc">earlier discussion of material balances</span></a> that the open species mole balance around component <span class="math notranslate nohighlight">\(i\)</span> in a steady-state system with species <span class="math notranslate nohighlight">\(\mathcal{M}\)</span>, streams <span class="math notranslate nohighlight">\(\mathcal{S}\)</span> and reactions <span class="math notranslate nohighlight">\(\mathcal{R}\)</span> is given by:</p>
<div class="math notranslate nohighlight">
\[\sum_{s\in\mathcal{S}}v_{s}\dot{n}_{si} + \sum_{j\in\mathcal{R}}\sigma_{ij}\dot{\epsilon}_{j} = 0\qquad\forall{i}\in\mathcal{M}\]</div>
<p>If we don’t have <a class="reference internal" href="chemical-kinetics.html"><span class="doc std std-doc">kinetic rate laws</span></a>, or we are not approaching this problem from the <a class="reference internal" href="chemical-equilibrium.html"><span class="doc std std-doc">equilibrium or energy perspective</span></a>, then we need some other way to estimate the extent of reaction <span class="math notranslate nohighlight">\(\dot{\epsilon}_{j}\)</span>.</p>
<div class="proof observation admonition" id="obs-simple-mole-based case">
<p class="admonition-title"><span class="caption-number">Observation 29 </span> (Mole based problem formulation)</p>
<section class="observation-content" id="proof-content">
<p>Suppose we have an open steady-state system with species <span class="math notranslate nohighlight">\(\mathcal{M}\)</span>, streams <span class="math notranslate nohighlight">\(\mathcal{S}\)</span> and reactions <span class="math notranslate nohighlight">\(\mathcal{R}\)</span>, but we have only a single stream entering (s = 1) and exiting (s = 2) the system. Further, suppose we want to maximize/minimize an objective of the form:</p>
<div class="math notranslate nohighlight">
\[\text{maximize/minimze}~\mathcal{O}=\sum_{i\in\mathcal{R}}c_{i}\dot{\epsilon}_{i}\]</div>
<p>In other words, we want to estimate the open extent such that some <em>technological</em> objective is met, e.g., we maximize the production of a valuable product. For the case of a single input and output, the open mole balance becomes:</p>
<div class="math notranslate nohighlight">
\[\dot{n}_{2,i} = \dot{n}_{1,i} + \sum_{j\in\mathcal{R}}\sigma_{ij}\dot{\epsilon}_{j}\qquad\forall{i}\in\mathcal{M}\]</div>
<p>Of course, when searching for the optimal set of reaction extents <span class="math notranslate nohighlight">\(\dot{\epsilon}_{j}\)</span> we have to select values that give physically realistic answers, e.g., we can’t have a negative mole flow rates: <span class="math notranslate nohighlight">\(\dot{n}_{i,j}\geq{0}\)</span> for all <span class="math notranslate nohighlight">\(i\)</span> and <span class="math notranslate nohighlight">\(j\)</span>. Thus, the species mole balances (which are linear) are constraints that make sure we select physically realistic values for the optimal open extent:</p>
<div class="math notranslate nohighlight">
\[\dot{n}_{1,i} + \sum_{j\in\mathcal{R}}\sigma_{ij}\dot{\epsilon}_{j}\geq{0}\qquad\forall{i}\in\mathcal{M}\]</div>
<p>Next, the <span class="math notranslate nohighlight">\(\dot{\epsilon}_{j}\)</span> terms, like concentration-based flux, must be bounded, i.e., we can’t have infinite values for the open reaction extent, nor can irreversible reactions be chosen to run backward. Thus, the extents are <em>bounded</em> from above and below:</p>
<div class="math notranslate nohighlight">
\[\mathcal{L}_{j}\leq\dot{\epsilon}_{j}\leq\mathcal{U}_{j}\qquad{j=1,2\dots,\mathcal{R}}\]</div>
<p>The <span class="math notranslate nohighlight">\(\mathcal{L}_{j}\)</span> and <span class="math notranslate nohighlight">\(\mathcal{U}_{j}\)</span> denote the lower and upper bounds that <span class="math notranslate nohighlight">\(\dot{\epsilon}_{j}\)</span> can take. Open extents <span class="math notranslate nohighlight">\(\dot{\epsilon}_{j}\)</span> are just reaction rates times the volume; the lower and upper bounds describe the permissible range we expect the rate <em>could</em> take.</p>
</section>
</div><p>Putting these ideas together for the general case of multiple input and output streams gives a different problem formulation to compute the flux through a reaction network using mole based units:</p>
<div class="proof definition admonition" id="defn-mol-based-units">
<p class="admonition-title"><span class="caption-number">Definition 39 </span> (Mole based flux problem)</p>
<section class="definition-content" id="proof-content">
<p>Suppose we have an open system with species <span class="math notranslate nohighlight">\(\mathcal{M}\)</span>, streams <span class="math notranslate nohighlight">\(\mathcal{S}\)</span> and reactions <span class="math notranslate nohighlight">\(\mathcal{R}\)</span>. Further, suppose we partition the stream set <span class="math notranslate nohighlight">\(\mathcal{S}\)</span> into streams entering the system <span class="math notranslate nohighlight">\(\mathcal{S}^{+}\)</span>, and streams leaving the system <span class="math notranslate nohighlight">\(\mathcal{S}^{-}\)</span>.</p>
<p>Then, the steady-state species mole balances are given by:</p>
<div class="math notranslate nohighlight">
\[\sum_{s\in\mathcal{S}^{+}}\dot{n}_{si} - \sum_{k\in\mathcal{S}^{-}}\dot{n}_{ki} + \sum_{j\in\mathcal{R}}\sigma_{ij}\dot{\epsilon}_{j} = 0\qquad\forall{i}\in\mathcal{M}\]</div>
<p>Finally, we know that <span class="math notranslate nohighlight">\(\dot{n}_{i,j}\geq{0}\)</span> for every <span class="math notranslate nohighlight">\(i\)</span> and <span class="math notranslate nohighlight">\(j\)</span>; species mole flows must be non-negative. Then, the (unknown) open extents <span class="math notranslate nohighlight">\(\dot{\epsilon}_{j}\)</span> are the solution of a linear programming problem in which the linear objective <span class="math notranslate nohighlight">\(\mathcal{O}\)</span>:</p>
<div class="math notranslate nohighlight">
\[\text{maximize/minimize}~\mathcal{O} = \sum_{j\in\mathcal{R}}c_{j}\dot{\epsilon}_{j}\]</div>
<p>is minimized (or maximized) subject to a collection of linear constraints:</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{eqnarray}
\sum_{j\in\mathcal{R}}\sigma_{ij}\dot{\epsilon}_{j}&amp;\geq&amp;{-\sum_{s\in\mathcal{S}^{+}}\dot{n}_{si}}\qquad\forall{i}\in\mathcal{M}\\
\sum_{k\in\mathcal{S}^{-}}\dot{n}_{ki}&amp;\geq&amp;{0}\qquad\forall{i}\in\mathcal{M}\\
\mathcal{L}_{j}&amp;\leq\dot{\epsilon}_{j}\leq&amp;\mathcal{U}_{j}\qquad\forall{j}\in\mathcal{R}
\end{eqnarray}\end{split}\]</div>
</section>
</div></section>
<section id="convex-pathway-analysis-advanced-topic">
<h4>Convex pathway analysis (advanced topic)<a class="headerlink" href="#convex-pathway-analysis-advanced-topic" title="Permalink to this headline">#</a></h4>
<p>The stoichiometric matrix <span class="math notranslate nohighlight">\(\mathbf{S}\)</span> is a <span class="math notranslate nohighlight">\(\mathcal{M}\times\mathcal{R}\)</span> array holding the
coefficients <span class="math notranslate nohighlight">\(\sigma_{ij}\)</span>; each row of <span class="math notranslate nohighlight">\(\mathbf{S}\)</span> describes a metabolite, while each column represents a possible chemical reaction. Thus, the stoichiometric matrix is a <em>digital</em> representation of the biochemical capabilities of a metabolic network.</p>
<p>When formulating the stoichiometric matrix, what should be included to describe known or unknown desired phenomena? For example, suppose we are interested in producing product <span class="math notranslate nohighlight">\(P\)</span> or making sense of a metabolic data set. Should every possible reaction or just some <em>subset</em> of all possible reactions be included in the stoichiometric matrix? The practical significance of this question is two-fold:</p>
<ul class="simple">
<li><p>Fewer material balances are better when describing the behavior of a metabolic network, and</p></li>
<li><p>Fewer chemical reactions (fluxes) result in less uncertainty when estimating the kinetic rates <span class="math notranslate nohighlight">\(\hat{r}_{j}\)</span> (bounds) or other constraints.</p></li>
</ul>
<p>Convex pathway analysis is an approach to catalog all possible steady-state pathways through a reaction network. In particular, convex decomposition approaches posit that any potential steady-state flux can be represented as the <a class="reference external" href="https://en.wikipedia.org/wiki/Convex_combination">convex combination</a> of a set of <a class="reference external" href="https://en.wikipedia.org/wiki/Convex_position">convex independent</a> basis vectors called <em>extreme pathways</em> <span class="math notranslate nohighlight">\(\mathbf{P}_{i}\)</span>:</p>
<div class="math notranslate nohighlight">
\[\mathbf{v} = \sum_{i=1}^{\mathcal{P}}\alpha_{i}\mathbf{P}_{i}\]</div>
<p>where <span class="math notranslate nohighlight">\(\mathcal{P}\)</span> denotes the number of basis pathways, and <span class="math notranslate nohighlight">\(\mathbf{P}_{i}\)</span> denotes the <span class="math notranslate nohighlight">\(\mathcal{R}\times{1}\)</span> extreme pathway basis vector. The term(s) <span class="math notranslate nohighlight">\(\alpha_{i}\geq{0}\)</span> denotes the weight of extreme pathway <span class="math notranslate nohighlight">\(i\)</span>.</p>
<p>Extreme pathways are <em>very difficult</em> to compute but can give super exciting insights into the operation and capabilities of a metabolic network.</p>
</section>
</section>
</section>
<hr class="docutils" />
<section id="summary">
<h2>Summary<a class="headerlink" href="#summary" title="Permalink to this headline">#</a></h2>
<p>This lecture introduced metabolic engineering and tools such as flux balance analysis to estimate the reaction rate through a (bio)chemical reaction network.</p>
<p>Flux balance analysis, written in either concentration or mole-based forms, is a linear programming problem in which the values for the unknown reaction rates or open reaction extents (which we collectively called <em>fluxes</em>) are estimated by minimizing (maximizing) the sum of the unknown fluxes subject to physical conservation constraints and constraints on the permissible values that the flux can take.  While the material we introduced was in the context of estimating unknown reaction rates, the ideas that underly flux balance analysis are more general; these same principles are used in various applications in many disciplines.</p>
<p>In this lecture, we:</p>
<ul class="simple">
<li><p>Introduced the <a class="reference internal" href="#content-references-theory-flux-balance-analysis"><span class="std std-ref">Theory of Flux Balance Analysis</span></a>, the <a class="reference internal" href="#content-references-flux-balance-analysis-traditional-structure"><span class="std std-ref">Traditional FBA problem structure</span></a> and an <a class="reference internal" href="#content-references-flux-balance-analysis-alternative-structure"><span class="std std-ref">Alternative approach</span></a> that used mole balances.</p></li>
</ul>
</section>
<section id="references">
<h2>References<a class="headerlink" href="#references" title="Permalink to this headline">#</a></h2>
<p>Background resources for metabolic engineering:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/2047876/">Bailey JE. Toward a science of metabolic engineering. Science. 1991 Jun 21;252(5013):1668-75. doi: 10.1126/science.2047876. PMID: 2047876.</a></p></li>
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/1904627/">Stephanopoulos G, Vallino JJ. Network rigidity and metabolic engineering in metabolite overproduction. Science. 1991 Jun 21;252(5013):1675-81. doi: 10.1126/science.1904627. PMID: 1904627.</a></p></li>
</ul>
<p>Background resources for biochemical network information:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://www.genome.jp/kegg/">Kanehisa M, Goto S. KEGG: kyoto encyclopedia of genes and genomes. Nucleic Acids Res. 2000 Jan 1;28(1):27-30. doi: 10.1093/nar/28.1.27. PMID: 10592173; PMCID: PMC102409.</a></p></li>
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/29447345/">Karp, Peter D et al. “The BioCyc collection of microbial genomes and metabolic pathways.” Briefings in bioinformatics vol. 20,4 (2019): 1085-1093. doi:10.1093/bib/bbx085</a></p></li>
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/26527724/">Gama-Castro, Socorro et al. “RegulonDB version 9.0: high-level integration of gene regulation, coexpression, motif clustering and beyond.” Nucleic acids research vol. 44,D1 (2016): D133-43. doi:10.1093/nar/gkv1156</a></p></li>
</ul>
<p>Background resources convex pathway analysis:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/15613397/">Bell SL, Palsson BØ. Expa: a program for calculating extreme pathways in biochemical reaction networks. Bioinformatics. 2005 Apr 15;21(8):1739-40. doi: 10.1093/bioinformatics/bti228. Epub 2004 Dec 21. PMID: 15613397.</a></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2909134/">Trinh CT, Wlaschin A, Srienc F. Elementary mode analysis: a useful metabolic pathway analysis tool for characterizing cellular metabolism. Appl Microbiol Biotechnol. 2009;81(5):813-826. doi:10.1007/s00253-008-1770-1</a></p></li>
<li><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4299494/">Bordbar A, Nagarajan H, Lewis NE, et al. Minimal metabolic pathway structure is consistent with associated biomolecular interactions. Mol Syst Biol. 2014;10(7):737. Published 2014 Jul 1. doi:10.15252/msb.20145243</a></p></li>
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/28369193/">Song HS, Goldberg N, Mahajan A, Ramkrishna D. Sequential computation of elementary modes and minimal cut sets in genome-scale metabolic networks using alternate integer linear programming. Bioinformatics. 2017 Aug 1;33(15):2345-2353. doi: 10.1093/bioinformatics/btx171. PMID: 28369193.</a></p></li>
</ul>
<p>Computational tools for working with flux balance analysis models:</p>
<ul class="simple">
<li><p><a class="reference external" href="https://pubmed.ncbi.nlm.nih.gov/30787451/">Heirendt, Laurent et al. “Creation and analysis of biochemical constraint-based models using the COBRA Toolbox v.3.0.” Nature protocols vol. 14,3 (2019): 639-702. doi:10.1038/s41596-018-0098-2</a></p></li>
</ul>
</section>
</section>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./chapter-3-dir"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
            </main>
            <footer class="footer-article noprint">
                
    <!-- Previous / next buttons -->
<div class='prev-next-area'>
    <a class='left-prev' id="prev-link" href="chemical-equilibrium.html" title="previous page">
        <i class="fas fa-angle-left"></i>
        <div class="prev-next-info">
            <p class="prev-next-subtitle">previous</p>
            <p class="prev-next-title">Introduction to Chemical Reaction Equilibrium</p>
        </div>
    </a>
    <a class='right-next' id="next-link" href="../References.html" title="next page">
    <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">References</p>
    </div>
    <i class="fas fa-angle-right"></i>
    </a>
</div>
            </footer>
        </div>
    </div>
    <div class="footer-content row">
        <footer class="col footer"><p>
  
    By Jeffrey D. Varner<br/>
  
      &copy; Copyright 2022.<br/>
</p>
        </footer>
    </div>
    
</div>


      </div>
    </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="../_static/scripts/pydata-sphinx-theme.js?digest=1999514e3f237ded88cf"></script>


  </body>
</html>